tests/psimulate/worker/test_vivarium_work_horse.py

```diff
@@ -48,3 +50,64 @@ def compare_dicts(test_dict, ref_dict):
     for test_dict in [sim_config, job_config]:
         for ref_dict in [initial_job_params, update_dict]:
             compare_dicts(test_dict, ref_dict)
+
+
+@pytest.mark.parametrize(
+    "make_dir,has_metadata_file,has_backup",
+    [(False, False, False), (True, False, False), (True, True, False), (True, True, True)],
+)
+def test_get_backup(tmp_path, make_dir, has_metadata_file, has_backup) -> None:
+    input_draw = 1
+    random_seed = 2
+    branch_configuration = {"branch_key": "branch_value"}
+    job_id = "job_id"
+    job_parameters = JobParameters(
+        model_specification=None,
+        branch_configuration=branch_configuration,
+        input_draw=input_draw,
+        random_seed=random_seed,
+        results_path="~/tmp",
+        backup_configuration={
+            "backup_dir": tmp_path / "backups",
+            "backup_metadata_path": tmp_path / "backups" / "backup_metadata.csv",
+        },
+        extras={},
+    )
+    if make_dir:
+        (tmp_path / "backups").mkdir(exist_ok=False)
+        if has_metadata_file:
+            metadata_draw = input_draw if has_backup else 7
+            metadata = pd.DataFrame(
+                {
+                    "input_draw": [metadata_draw],
+                    "random_seed": [random_seed],
+                    "job_id": job_id,
+                    "branch_key": ["branch_value"],
+                }
+            )
+            metadata.to_csv(tmp_path / "backups" / "backup_metadata.csv", index=False)
+
+    if make_dir and has_metadata_file and has_backup:
+        pickle_path = tmp_path / "backups" / f"{job_id}.pkl"
+        pickle = [1, 2, 3, 4, 5]
+        with open(pickle_path, "wb") as f:
+            dill.dump(pickle, f)
+        backup = get_backup(job_parameters)
+        assert backup == pickle
+
+    else:
+        backup = get_backup(job_parameters)
+        assert not backup
+
+
+def test_remove_backups(mocker, tmp_path) -> None:
+    mocker.patch(
+        "vivarium_cluster_tools.psimulate.worker.vivarium_work_horse.get_current_job",
+        return_value=mocker.Mock(id="job_id"),
+    )
+    # touch a file
+    (tmp_path / "job_id.pkl").touch()
+    assert (tmp_path / "job_id.pkl").exists()
+    # remove the file
+    remove_backups(tmp_path)
+    assert not (tmp_path / "job_id.pkl").exists()
```

Comment:
I'm not sure how you would do it but it seems useful to test we are in fact loading from the most previous back up point however that would require you to both fail a job and restart in in a test.

Reply:
I added to this test to load from potentially multiple existing backups. For unit tests, I think we don't necessarily need to kill an active job and restart it--we can test it from either end "as though" a job failed and we're picking it back up.

It would be great to have some e2e tests for something like this, but i think that would have to be on Jenkins.